src/puppetlabs/trapperkeeper/services/authorization/authorization_core.clj

```diff
@@ -118,8 +125,16 @@
   (when-not (string? (:path (:match-request rule)))
     (throw (IllegalArgumentException.
             (str "The path set in the authorization rule specified as "
-                 (pprint-rule (:path (:match-request rule))) " is invalid. "
+                 (pprint-rule rule) " is invalid. "
                  "It should be a string."))))
+  (when (and (not (nil? (:method rule)))
```

Comment:
The `:method` is actually part of the `match-request` with the rule - like `path` - and not a top-level member of the rule.  So I don't think this is validating what it needs to be.

Also, will this work if the value for `method` is a vector?  In that case, I think we'd want to be validating that each of the values in the vector is one of the `valid-methods`.